### PR TITLE
EVPN VXLAN DB support

### PR DESCRIPTION
**Pull request by srj102**
Added support for new VXLAN DBs as described in the PR Azure/SONiC#437

### CODE REVIEW

File: common/schema.h [hunk 1]
```diff
@@ -251,6 +263,10 @@ namespace swss {
 #define STATE_NEIGH_RESTORE_TABLE_NAME              "NEIGH_RESTORE_TABLE"
 #define STATE_MIRROR_SESSION_TABLE_NAME             "MIRROR_SESSION_TABLE"
 #define STATE_VXLAN_TABLE_NAME                      "VXLAN_TABLE"
+#define STATE_VXLAN_TUNNEL_TABLE_NAME               "VXLAN_TUNNEL_TABLE"
+
+
+#define STATE_TUNNEL_VLAN_MAP_TABLE_NAME            "TUNNEL_VLAN_MAP_TABLE"
```
Comment (prsunny) [inline common/schema.h:269]: Can you rename this to VXLAN_TUNNEL_MAP.. to be in consistent with CFG_VXLAN_TUNNEL_MAP_TABLE_NAME

File: common/schema.h [hunk 2]
```diff
@@ -210,12 +217,17 @@ namespace swss {
 
 #define CFG_VXLAN_TUNNEL_TABLE_NAME                 "VXLAN_TUNNEL"
 #define CFG_VXLAN_TUNNEL_MAP_TABLE_NAME             "VXLAN_TUNNEL_MAP"
+#define CFG_VXLAN_EVPN_NVO_TABLE_NAME               "EVPN_NVO"
```
Comment (prsunny) [inline common/schema.h:220]: Please rename to VXLAN_EVPN_NVO to be consistent

File: common/schema.h [hunk 3]
```diff
@@ -52,6 +56,9 @@ namespace swss {
 #define APP_NAT_GLOBAL_TABLE_NAME       "NAT_GLOBAL_TABLE"
 #define APP_NAPT_POOL_IP_TABLE_NAME     "NAPT_POOL_IP_TABLE"
 
+
+#define APP_NEIGH_SUPPRESS_VLAN_TABLE_NAME   "SUPPRESS_VLAN_NEIGH_TABLE"
```
Comment (prsunny) [inline common/schema.h:60]: This is duplicate, already defined above

File: common/schema.h [hunk 4]
```diff
@@ -38,6 +38,10 @@ namespace swss {
 #define APP_VXLAN_VRF_TABLE_NAME          "VXLAN_VRF_TABLE"
 #define APP_VXLAN_TUNNEL_MAP_TABLE_NAME   "VXLAN_TUNNEL_MAP_TABLE"
 #define APP_VXLAN_TUNNEL_TABLE_NAME       "VXLAN_TUNNEL_TABLE"
+#define APP_VXLAN_FDB_TABLE_NAME          "VXLAN_FDB_TABLE"
+#define APP_VXLAN_REMOTE_VNI_TABLE_NAME   "VXLAN_REMOTE_VNI_TABLE"
+#define APP_EVPN_NVO_TABLE_NAME           "EVPN_NVO_TABLE"
```
Comment (prsunny) [inline common/schema.h:43]: Please rename to VXLAN_EVPN_NVO.. as you've defined for config - CFG_VXLAN_EVPN_NVO_TABLE_NAME

File: common/schema.h [hunk 5]
```diff
@@ -210,12 +217,17 @@ namespace swss {
...
+#define CFG_VXLAN_EVPN_NVO_TABLE_NAME               "EVPN_NVO"
 #define CFG_VNET_TABLE_NAME                         "VNET"
 #define CFG_NEIGH_TABLE_NAME                        "NEIGH"
+#define CFG_NEIGH_SUPPRESS_VLAN_TABLE_NAME          "SUPPRESS_VLAN_NEIGH"
 
 #define CFG_VNET_RT_TABLE_NAME                      "VNET_ROUTE"
 #define CFG_VNET_RT_TUNNEL_TABLE_NAME               "VNET_ROUTE_TUNNEL"
 
+
+#define CFG_NEIGH_SUPPRESS_VLAN_TABLE_NAME          "SUPPRESS_VLAN_NEIGH"
```
[the remark this answers is not in the document]
Comment (srj102) [reply common/schema.h:229]: will do.

File: common/schema.h [hunk 6]
```diff
@@ -251,6 +263,10 @@ namespace swss {
 #define STATE_NEIGH_RESTORE_TABLE_NAME              "NEIGH_RESTORE_TABLE"
 #define STATE_MIRROR_SESSION_TABLE_NAME             "MIRROR_SESSION_TABLE"
 #define STATE_VXLAN_TABLE_NAME                      "VXLAN_TABLE"
+#define STATE_VXLAN_TUNNEL_TABLE_NAME               "VXLAN_TUNNEL_TABLE"
+
+
```
[the remark this answers is not in the document]
Comment (srj102) [reply common/schema.h:268]: will do

File: common/schema.h (hunk 2 above)
Comment (srj102) [reply common/schema.h:220]: will do

File: common/schema.h (hunk 3 above)
Comment (srj102) [reply common/schema.h:60]: will do

File: common/schema.h (hunk 4 above)
Comment (srj102) [reply common/schema.h:43]: will do

File: common/schema.h (hunk 1 above)
Comment (tapashdas) [reply common/schema.h:269]: This table is related to Neighbor suppression changes. Will modify the name.
Will this name "STATE_NEIGH_SUPRESS_VLAN_TABLE_NAME" be fine ?

Comment (prsunny) [reply common/schema.h:269]: ok

**Comment by prsunny** [comment]
@srj102 , can you rebase and address any outstanding comments.

**Comment by prsunny** [comment]
@srj102 , lets get this PR marked for ready. Also, have only the evpn attributes in this PR. I see some changes related to warmboot etc. That can be in another PR.

File: common/schema.h [hunk 7]
```diff
@@ -38,6 +38,10 @@ namespace swss {
 #define APP_VXLAN_VRF_TABLE_NAME          "VXLAN_VRF_TABLE"
 #define APP_VXLAN_TUNNEL_MAP_TABLE_NAME   "VXLAN_TUNNEL_MAP_TABLE"
 #define APP_VXLAN_TUNNEL_TABLE_NAME       "VXLAN_TUNNEL_TABLE"
+#define APP_VXLAN_FDB_TABLE_NAME          "VXLAN_FDB_TABLE"
+#define APP_VXLAN_REMOTE_VNI_TABLE_NAME   "VXLAN_REMOTE_VNI_TABLE"
+#define APP_EVPN_NVO_TABLE_NAME           "VXLAN_EVPN_NVO_TABLE"
```
Comment (prsunny) [inline common/schema.h:43]: Based on previous comment, it should have been APP_EVPN_NVO_TABLE_NAME  -> APP_VXLAN_EVPN_NVO_TABLE_NAME to be consistent with CFG_VXLAN_EVPN_NVO_TABLE_NAME

File: common/schema.h [hunk 8]
```diff
@@ -251,6 +257,8 @@ namespace swss {
 #define STATE_NEIGH_RESTORE_TABLE_NAME              "NEIGH_RESTORE_TABLE"
 #define STATE_MIRROR_SESSION_TABLE_NAME             "MIRROR_SESSION_TABLE"
 #define STATE_VXLAN_TABLE_NAME                      "VXLAN_TABLE"
+#define STATE_VXLAN_TUNNEL_TABLE_NAME               "VXLAN_TUNNEL_TABLE"
+#define STATE_NEIGH_SUPPRESS_VLAN_TABLE_NAME        "NEIGH_SUPPRESS_VLAN_MAP_TABLE"
```
Comment (prsunny) [inline common/schema.h:261]: I think there is a mismatch here w.r.t APP DB -  "NEIGH_SUPPRESS_VLAN_MAP_TABLE" -> "SUPPRESS_VLAN_NEIGH_TABLE". Can you please correct this one?

Comment (tapashdas) [reply common/schema.h]: Done

**Pull request merged**